src/Adapter/InMemoryAdapter.php

```diff
@@ -34,11 +33,21 @@ class InMemoryAdapter implements Adapter
      */
     protected $streams;
 
+    public function fetchStreamMetadata(StreamName $streamName): ?array
+    {
+        if (isset($this->streams[$streamName->toString()]['metadata'])) {
+            return $this->streams[$streamName->toString()]['metadata'];
+        }
+
+        return null;
```

Comment:
similar as `return $this->streams[$streamName->toString()]['metadata'] ?? null;`